operator/pkg/resources/statefulset_scale.go

```diff
@@ -71,85 +74,135 @@ const (
+		log.V(logger.DebugLevel).Info("No scaling changes required for this nodepool", "replicas", *r.nodePool.Replicas, "spec replicas", *r.LastObservedState.Spec.Replicas) // No changes to replicas, we do nothing here
 
-		// We care about upscaling only when the cluster is moving off 1 replica, which happen e.g. at cluster startup
-		if r.pandaCluster.Status.CurrentReplicas == 1 {
-			r.logger.Info("Waiting for first node to form a cluster before upscaling")
-			formed, err := r.isClusterFormed(ctx)
-			if err != nil {
-				return err
-			}
-			if !formed {
-				return &RequeueAfterError{
-					RequeueAfter: wait.Jitter(r.decommissionWaitInterval, decommissionWaitJitterFactor),
-					Msg:          fmt.Sprintf("Waiting for cluster to be formed before upscaling to %d replicas", *r.pandaCluster.Spec.Replicas),
-				}
-			}
-			r.logger.Info("Initial cluster has been formed")
-		}
+		return nil
+	}
 
-		// Upscaling request: this is already handled by Redpanda, so we just increase status currentReplicas
-		return setCurrentReplicas(ctx, r, r.pandaCluster, *r.pandaCluster.Spec.Replicas, r.logger)
+	if npCurrentReplicas == 0 {
+		// we 're done here
+		return nil
 	}
 
 	// User required replicas is lower than current replicas (currentReplicas): start the decommissioning process
-	r.logger.Info("Downscaling cluster", "replicas", *r.pandaCluster.Spec.Replicas)
+	r.logger.Info("Downscaling cluster", "nodepool_replicas", ptr.Deref(r.nodePool.Replicas, 0), "nodepool_current_replicas", npCurrentReplicas)
 
-	targetOrdinal := r.pandaCluster.Status.CurrentReplicas - 1 // Always decommission last node
+	targetOrdinal := npStatus.CurrentReplicas - 1 // Always decommission last node
 	targetBroker, err := r.getBrokerIDForPod(ctx, targetOrdinal)
 	if err != nil {
 		return fmt.Errorf("error getting broker ID for pod with ordinal %d when downscaling cluster: %w", targetOrdinal, err)
 	}
-	nonExistantBroker := int32(-1)
+
 	if targetBroker == nil {
 		// The target pod isn't in the broker list. Just select a non-existing broker for decommission so the next
 		// reconcile loop will succeed.
-		targetBroker = &nonExistantBroker
-	}
-	log.WithValues("ordinal", targetOrdinal, "node_id", targetBroker).Info("start decommission broker")
-	r.pandaCluster.SetDecommissionBrokerID(targetBroker)
-	err = r.Status().Update(ctx, r.pandaCluster)
-	if err != nil {
-		return err
-	}
-	if *targetBroker == nonExistantBroker {
+		// Previously, this was accepted, and status.decommissionBrokerID was cleared.
+		// However, this is dangerous.
+		// Broker ID could be missing for a variety of reasons? If we just ignore this, and remove the pod anyway and skip decom'ing..we may remove pod w/ ghost broker left.
+		// Instead, we only decommission, if both pod and brokerID are present. This effectively means, our "state machine" requires upscaling to have finished (pod started + broker registered), before we consider downscaling.
```

Comment:
Could you correct the whole comment as the first 2 lines are left from previous logic.

It would be good to elaborate about dangerous scenarios.

```
if both pod and brokerID are present.
```

I would change to indicate that BrokerID is an annotation within Pod metadata.